src/libpriv/rpmostree-importer.c

```diff
@@ -1030,3 +1030,94 @@ rpmostree_importer_get_header_sha256 (RpmOstreeImporter *self)
 {
   return self->hdr_sha256;
 }
+
+
+static gboolean
+import_local_rpm (OstreeRepo    *repo,
+                  int           *fd,
+                  char         **sha256_nevra,
+                  GCancellable  *cancellable,
+                  GError       **error)
+{
+  /* let's just use the current sepolicy -- we'll just relabel it if the new
+   * base turns out to have a different one */
+  glnx_autofd int rootfs_dfd = -1;
+  if (!glnx_opendirat (AT_FDCWD, "/", TRUE, &rootfs_dfd, error))
+    return FALSE;
+  g_autoptr(OstreeSePolicy) policy = ostree_sepolicy_new_at (rootfs_dfd, cancellable, error);
+  if (policy == NULL)
+    return FALSE;
+
+  g_autoptr(RpmOstreeImporter) unpacker = rpmostree_importer_new_take_fd (fd, repo, NULL, 0, policy, error);
+  if (unpacker == NULL)
+    return FALSE;
+
+  if (!rpmostree_importer_run (unpacker, NULL, cancellable, error))
+    return FALSE;
+
+  g_autofree char *nevra = rpmostree_importer_get_nevra (unpacker);
+  *sha256_nevra = g_strconcat (rpmostree_importer_get_header_sha256 (unpacker),
+                               ":", nevra, NULL);
+
+  return TRUE;
+}
+
+static void
+ptr_close_fd (gpointer fdp)
+{
+  int fd = GPOINTER_TO_INT (fdp);
+  glnx_close_fd (&fd);
+}
+
+/* GUnixFDList doesn't allow stealing individual members */
+static GPtrArray *
+unixfdlist_to_ptrarray (GUnixFDList *fdl)
+{
+  gint len;
+  gint *fds = g_unix_fd_list_steal_fds (fdl, &len);
+  GPtrArray *ret = g_ptr_array_new_with_free_func ((GDestroyNotify)ptr_close_fd);
+  for (int i = 0; i < len; i++)
+    g_ptr_array_add (ret, GINT_TO_POINTER (fds[i]));
+  return ret;
+}
+
+gboolean
+rpmostree_importer_import_many_local_rpms (OstreeRepo    *repo,
```

Comment:
I think given that this in in the importer now, maybe just `rpmostree_importer_import_rpms` ?

Reply:
Let's also make this code move a separate prep commit? I think that'll clean up the history nicely make it easier to review the critical changes.

Reply:
> I think given that this in in the importer now, maybe just `rpmostree_importer_import_rpms` ?

Sounds good.  I was trying to think of a better but I am terrible with naming.



> Let's also make this code move a separate prep commit? I think that'll clean up the history nicely make it easier to review the critical changes.

Will do.